### PR TITLE
Fix wallmounted_to_facedir redefinition

### LINKED ISSUE #6739: MSVC compiler error "wallmounted_to_facedir": redefinition
"wallmounted_to_facedir" is declared as extern in minetest\src\util\directiontables.h(34) but it isn't actually extern. 

Original localized error message:
```
2>\Minetest\src\util\directiontables.cpp(113): error C2370: "wallmounted_to_facedir": Neudefinition; unterschiedliche Speicherklassen
2>\minetest\src\util\directiontables.h(34): note: Siehe Deklaration von "wallmounted_to_facedir"
```

### PR DESCRIPTION
**Pull request by adrido**
Fixes #6739

### CODE REVIEW

File: src/util/directiontables.h [hunk 1]
```diff
@@ -31,7 +31,14 @@ extern const v3s16 g_26dirs[26];
 // 26th is (0,0,0)
 extern const v3s16 g_27dirs[27];
 
-extern const u8 wallmounted_to_facedir[6];
+constexpr const u8 wallmounted_to_facedir[6] = {
```
Comment (SmallJoker) [inline src/util/directiontables.h]: It's just a header, so `extern constexpr const u8 wallmounted_to_facedir[6];` isn't valid?

Comment (adrido) [reply src/util/directiontables.h]: if its a constexpr it have to initialized immediately, so its invalid.

**Comment by paramat** [comment]
More input needed on this.

**Comment by SmallJoker** [comment]
#6785 seems to be the better approach to fix this issue. Headers should not contain initialisations.

**Pull request closed without merging**